core/types/calls.cc

```diff
@@ -1626,12 +1626,30 @@ DispatchResult badMetaTypeCall(const GlobalState &gs, const DispatchArgs &args,
                                const TypePtr &wrapped) {
     if (auto e = gs.beginError(errLoc, errors::Infer::MetaTypeDispatchCall)) {
         e.setHeader("Call to method `{}` on `{}` mistakes a type for a value", args.name.show(gs), wrapped.show(gs));
+
+        if (isa_type<SelfTypeParam>(wrapped)) {
+            auto selfTypeParam = cast_type_nonnull<SelfTypeParam>(wrapped);
+            if (selfTypeParam.definition.isTypeMember()) {
```

Comment:
```suggestion
        if (auto *selfTypeParam = cast_type<SelfTypeParam>(wrapped)) {
            if (selfTypeParam->definition.isTypeMember()) {
```

Reply:
This type is inlined so it didn’t let me write this.